pkg/provider/aws/secretsmanager/secretsmanager.go

```diff
@@ -631,10 +631,17 @@ func (sm *SecretsManager) constructSecretValue(ctx context.Context, ref esv1beta
 	}
 	secretOut, err := sm.client.GetSecretValue(getSecretValueInput)
 	metrics.ObserveAPICall(constants.ProviderAWSSM, constants.CallAWSSMGetSecretValue, err)
-	var nf *awssm.ResourceNotFoundException
+	var (
+		nf *awssm.ResourceNotFoundException
+		ie *awssm.InvalidRequestException
+	)
 	if errors.As(err, &nf) {
 		return nil, esv1beta1.NoSecretErr
 	}
 
+	if errors.As(err, &ie) && strings.Contains(ie.Error(), "was marked for deletion") {
```

Comment:
Unfortunately there is no other indication then the error message about the state of the secret. So there was nothing else I could "check".